### PR TITLE
Checks for Voxel Sign when format is Default

### LINKED ISSUE #44: nii2minc should not allow launching a task with one selection box left to default
There is a selection box in the nii2minc task view code that says:

```
 Write voxel data in type [selection box]
```

There are three values in it, but the task code allows launching a task with the blank value still selected, which crashes on the bourreau side with

```
 CbrainError: Unexpected voxel int sign
```

Adjust after_form().

### PR DESCRIPTION
**Pull request by crocodoyle**
Fixes #44. Also updated regex for security

### CODE REVIEW

**Comment by prioux** [comment]
Testing webhooks.

File: cbrain_task/nii2mnc/portal/nii2mnc.rb [hunk 1]
```diff
@@ -73,7 +73,7 @@ def after_form #:nodoc:
     params = self.params
     cb_error "Missing voxel type"     if params[:voxel_type].blank?
     cb_error "Missing space ordering" if params[:space_ordering].blank?    && !self.tool_config.is_at_least_version("2.0.0")
-    cb_error "Missing voxel int sign" if params[:voxel_int_signity].blank? && params[:voxel_type] =~ /^(byte|short|int)$/
+    cb_error "Missing voxel int sign" if params[:voxel_int_signity].blank? && params[:voxel_type] =~ /\A(default|byte|short|int)\z/
```
Comment (prioux) [inline cbrain_task/nii2mnc/portal/nii2mnc.rb]: The logic makes no sense? The if condition will never be true, it says if

* the value is blank

*and*

* the value is a string, one of default, byte, short or int

Comment (prioux) [inline cbrain_task/nii2mnc/portal/nii2mnc.rb]: Shouldn't this be

```ruby
cb_error "Missing voxel int sign" if params[:voxel_int_signity].blank? || params[:voxel_type] !~ /\A(default|byte|short|int)\z/
```

Comment (prioux) [inline cbrain_task/nii2mnc/portal/nii2mnc.rb]: (Two changes: `&&` becomes `||` and the match operator `=~` becomes non-match `!~`

Comment (crocodoyle) [inline cbrain_task/nii2mnc/portal/nii2mnc.rb]: The logic makes sense. There are two parameters at play here.

1. voxel_type
2. voxel_int_signity

If voxel_type is possibly an integer, then voxel_int_signity cannot be blank.

Comment (prioux) [inline cbrain_task/nii2mnc/portal/nii2mnc.rb]: My bad. You're totally right. I had not noticed that the symbols were different. It happens so often that we test something with

```ruby
blahblah if xyz.blank? || xyz.has_bad_property
```

that I didn't notice we were not actually checking the same xyz.

**Comment by prioux** [comment]
Man there were more discussions here than actual code modified. :-)

**Pull request merged**